Sources/Build/ToolProtocol.swift

```diff
@@ -61,22 +61,23 @@ struct SwiftcTool: ToolProtocol {
         // For C family targets Swift needs dynamic libraries to be able to interpolate.
         // We implicitly create dynamic libs for all C targets ie ClangModules, add
         // input to the product and not the module for ClangModules.
-        return module.recursiveDependencies.map{ module in
+        return module.recursiveDependencies.flatMap{ module in
             switch module {
             case let module as SwiftModule:
-                return module.targetName
+                return SwiftcTool(module: module, prefix: prefix, otherArgs: [], executable: executable, conf: conf).moduleOutputPath
```

Comment:
Yeah we need better infra for build metadata which needs to be accessed from different places